addons/account/models/account_bank_statement.py

```diff
@@ -778,27 +639,37 @@ def _check_amounts_currencies(self):
     # LOW-LEVEL METHODS
     # -------------------------------------------------------------------------
 
+    def new(self, values=None, origin=None, ref=None):
+        st_line = super().new(values, origin, ref)
+        if not st_line.journal_id:  # might not be computed because declared by inheritance
+            st_line.move_id._compute_journal_id()
+        return st_line
+
     @api.model_create_multi
     def create(self, vals_list):
         # OVERRIDE
         counterpart_account_ids = []
 
         for vals in vals_list:
-            statement = self.env['account.bank.statement'].browse(vals['statement_id'])
-            if statement.state != 'open' and self._context.get('check_move_validity', True):
-                raise UserError(_("You can only create statement line in open bank statements."))
+            if 'statement_id' in vals:
+                statement = self.env['account.bank.statement'].browse(vals['statement_id'])
+
+                if vals.get('journal_id') and statement.journal_id and statement.journal_id.id != vals['journal_id']:
+                    # most common case, create st_line by code with default journal is not in the context,
+                    # default statement does not take journal into account because it does not have access to
+                    # the vals, it is set in the default_get and on_change is not called.
+                    del vals['statement_id']
```

Comment:
simplify the code and always take the journal from the statement

Reply:
Are you sure about this? If the statement is not comply with the journal things would be weired, for example if the user saves the default journal in `ir.default`